repositories.bzl

```diff
@@ -108,6 +108,8 @@ cc_library(
 #
 ISTIO_API = "820986f2947c3f83154cf3f157d6145bb584830b"
 ISTIO_API_SHA256 = "453bf2257291ccd831b6fbdef350cb8a7d8f80a68f0a83beb024dc7cd64a4b95"
+GOGOPROTO_RELEASE = "1.2.1"
+GOGOPROTO_SHA256 = "99e423905ba8921e86817607a5294ffeedb66fdd4a85efce5eb2848f715fdb3a"
```

Comment:
This is already available as `@com_github_gogo_protobuf`, since Envoy API depends on it.

Is there any reason why we cannot use that?

Reply:
I got some error when using gogo proto library from upstream (`@com_github_gogo_protobuf//:gogo_proto`). It seems native `cc_proto_library` here is conflicting with `cc_proto_library` from google protobuf library, which is used by upstream gogo protobuf. @kyessenov Mentioned some reason related to python proto rule that blocks upstream gogoproto to use the native rule.

> 
> Label: @com_github_gogo_protobuf//:gogo_proto, @com_github_gogo_protobuf//:gogo_proto_cc_genproto
> RuleClass: proto_library rule, proto_gen rule
> Configuration: d609ccd60432de4740da0b6be20cd2eb
> Mnemonic: GenProto, ProtoCompile
> Action key: dc770c17fb46fb588e3bbfe3a573e508, de403051786b671b74f0f00bd261b57f
> Progress message: Generating C++ proto_library @com_github_gogo_protobuf//:gogo_proto, ProtoCompile external/com_github_gogo_protobuf/gogoproto/gogo.pb.h
....
> Owner information: []#BazelCcProtoAspect BuildConfigurationValue.Key[d609ccd60432de4740da0b6be20cd2eb] @com_github_gogo_protobuf//:gogo_proto BuildConfigurationValue.Key[d609ccd60432de4740da0b6be20cd2eb] false {}, @com_github_gogo_protobuf//:gogo_proto_cc_genproto BuildConfigurationValue.Key[d609ccd60432de4740da0b6be20cd2eb] false
> MandatoryInputs: Attempted action contains artifacts not in previous action (first 5): 
> 	external/com_google_protobuf/google/protobuf/descriptor.proto
> Previous action contains artifacts not in attempted action (first 5): 
> 	external/com_google_protobuf/src/google/protobuf/any.proto
> 	external/com_google_protobuf/src/google/protobuf/api.proto
> 	external/com_google_protobuf/src/google/protobuf/compiler/plugin.proto
> 	external/com_google_protobuf/src/google/protobuf/descriptor.proto
> 	external/com_google_protobuf/src/google/protobuf/duration.proto
> Outputs: Attempted action contains artifacts not in previous action (first 5): 
> 	external/com_github_gogo_protobuf/gogoproto/gogo.pb.h
> 	external/com_github_gogo_protobuf/gogoproto/gogo.pb.cc
> Previous action contains artifacts not in attempted action (first 5): 
> 	external/com_github_gogo_protobuf/gogoproto/gogo.pb.h
> 	external/com_github_gogo_protobuf/gogoproto/gogo.pb.cc
>

Reply:
Argh, OK.

Reply:
It looks that we cannot have 2 copies of `@com_github_gogo_protobuf` after all.

I think that you need to import the local version as `@com_github_gogo_protobuf` so that it shadows the version from upstream and copy required build targets from their `BUILD` file.